src/app/js/formats/streamgraph/Streamgraph.js

```diff
@@ -0,0 +1,410 @@
+import React, { PureComponent } from 'react';
+import { connect } from 'react-redux';
+import compose from 'recompose/compose';
+import * as d3 from 'd3';
+
+import { getCssToString, zoomFunction, distinctColors, transformDataIntoMapArray, randomId, getMinMaxValue } from './utils';
+import injectData from '../injectData';
+import exportableToPng from '../exportableToPng';
+
+const styles = {
+    divContainer: {
+        overflow: 'hidden',
+        position: 'relative',
+    },
+    tooltip: {
+        marginTop: 900,
+    },
+    vertical: {
+        marginTop: 60,
+        height: 190,
+        pointerEvents: 'none',
+    },
+    legend: {
+        position: 'relative',
+    },
+    legendItem: {
+        marginTop: 2,
+    },
+    legentItemText: {
+        marginLeft: 5,
+    },
+    legendButton: {
+        height: 15,
+        width: 15,
+    },
+};
+
+class Streamgraph extends PureComponent {
+
+    constructor(props) {
+        super(props);
+        this.state = {
+            width: 800,
+            height: 300
+        }
+
+        this.uniqueId = randomId(10);
+        this.divContainer = "divContainer" + this.uniqueId;
+        this.svgContainer = "svgContainer" + this.uniqueId;
+        this.anchor = "anchor" + this.uniqueId;
+
+        this.xAxisScale;
+        this.yAxisScale;
+        this.xAxis;
+        this.yAxisL;
+        this.yAxisR;
+        this.gx;
+        this.gyr;
+        this.gyl;
+        this.streams;
+
+        //this.draft = draft.bind(this);
+        this.zoomFunction = zoomFunction.bind(this);
+    }
+
+    setGraph() {
+        let [valuesObjectsArray, valuesArray, minDate, maxDate, nameList] = transformDataIntoMapArray(this.props.formatData);
+        const divContainerWidth = document.getElementById(this.divContainer).clientWidth;
+        let { height } = this.state;
+        let width = divContainerWidth;
+        const svgHeight = height;
+        const svgWidth = width;
+
+        const d3DivContainerId = "d3DivContainer" + this.uniqueId;
+        const divContainer = d3.select(this.refs.divContainer)
+        const d3DivContainer = divContainer.append("div")
+            .attr("id", d3DivContainerId);
+        const svgViewport = d3.select(this.refs.anchor);
+        let svgContainerDiv = d3.select(this.refs.svgContainer);
+        let componentContext = this;
+
+        // ===========================================================================================
+        // Set all the variables 
```

Comment:
You are forced to use this kind of comments because the `setGraph` function is HUGE.
Please, split it into multiple well named functions so that you don't have to write this kind of comments.